app/routes/surveys/components/SubmissionEditor.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import styles from './surveys.css';
+import React from 'react';
+import LoadingIndicator from 'app/components/LoadingIndicator';
+import { Field } from 'redux-form';
+import Button from 'app/components/Button';
+import { TextArea, RadioButton, CheckBox } from 'app/components/Form';
+import { createValidator, required } from 'app/utils/validation';
+import { reduxForm } from 'redux-form';
+import type { SurveyEntity } from 'app/reducers/surveys';
+import { Content, ContentHeader } from 'app/components/Content';
+import { Link } from 'react-router';
+
+type Props = {
+  survey: SurveyEntity,
+  submitting: boolean,
+  handleSubmit: ((Object) => Promise<*>) => void,
+  autoFocus: any,
+  fetching: boolean,
+  submitFunction: (Object, ?number) => Promise<*>
+};
+
+const SubmissionEditor = ({
+  survey,
+  fetching,
+  submitting,
+  handleSubmit,
+  submitFunction
+}: Props) => {
+  const onSubmit = (formContent: Object) => {
+    const toSubmit = {
+      ...formContent,
+      user: formContent.user && formContent.user.id,
+      surveyId: survey.id,
+
+      answers: formContent.answers
+        .map((answer, i) => {
+          const question = survey.questions[i];
+          const selected = answer.selectedOptions || [];
+          const selectedOptions =
+            question.questionType === 1
+              ? selected.map(Number)
+              : selected
+                  .map(
+                    (optionSelected, j) =>
+                      optionSelected && question.options[j].id
+                  )
+                  .filter(option => option);
+
+          return {
+            ...answer,
+            question: question.id,
+            selectedOptions,
+            answerText: answer.answerText || ''
+          };
+        })
+        .filter(answer => answer)
+    };
+
+    return submitFunction(toSubmit);
+  };
+
+  if (fetching || !survey || !survey.event || !survey.event.cover) {
+    return <LoadingIndicator loading />;
```

Comment:
Not a big deal, but this could be replaced with the `loggedIn` HOC inside the route.

Reply:
It's kind of a workaround, since it seemed like `survey` gets populated as an object without these values in early renders. I don't know if the Route-based `loadingIndicator` handles that. This solution doesn't look pretty though, I'll give you that, it was simply my response to getting `survey.event is not defined` despite having checked that `survey` is defined.

Reply:
It should be possible to do it like this:
https://github.com/webkom/lego-webapp/blob/46191ef7cae05cb65e60362b5dba8689a1ecb423/app/routes/admin/email/EmailUsersRoute.js#L22
```jsx
loadingIndicator(['survey.event.cover', 'notFetching']) 
```